application/controllers/admin/usergroups.php

```diff
@@ -121,10 +121,9 @@ public function delete()
         if (Permission::model()->hasGlobalPermission('usergroups', 'delete')) {
             $ugid = Yii::app()->request->getPost("ugid");
             if (!empty($ugid) && ($ugid > -1)) {
-                $result = UserGroup::model()->requestEditGroup($ugid, Yii::app()->session["loginID"]);
-                if ($result->count() > 0) {
-                    $delquery_result = UserGroup::model()->deleteGroup($ugid, Yii::app()->session["loginID"]);
```

Comment:
The permission checking done in `UserGroup::model()->deleteGroup()` is done anyway in `UserGroup::model()->requestEditGroup()` and even that should not be in model. Added the `@deperecated` to `UserGroup::model()->deleteGroup()` (was also only isage I found)

Reply:
@Shnoulle I just wanted a second pair of eyes to have a look and github suggested you :)

and yes agree totally about 1 & 2
3 sometimes maybe (if 2 not possible).

Reply:
Since we don't have same system in Group than in Survey : yes, this don't break system.